tests/test_train/test_hooks.py

```diff
@@ -244,6 +243,29 @@ def review(self, inputs, outputs):
             }
 
 
+def test_summary_hook_create_snapshot_flag():
+    class Model(DummyModel):
+
+        def __init__(self, validation_losses, exp_dir, optimizer):
+            super().__init__(validation_losses, exp_dir, optimizer)
+            self.create_snapshot_log = []
+
+        def review(self, example, output):
+            self.create_snapshot_log.append(self.create_snapshot)
+            return super().review(example, output)
+
+    ds = [0., 1., 2.]
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        optimizer = pt.optimizer.Adam()
+        model = Model([1, 2, 3] * 10, tmp_dir, optimizer)
+        trainer = pt.Trainer(
+            model, tmp_dir, optimizer, stop_trigger=(10, 'epoch'),
+            summary_trigger=(1, 'epoch')
+        )
```

Comment:
Could you add a second test where you also add `trainer.register_validation_hook`?

Reply:
Done